file/file_prefetch_buffer.h

```diff
@@ -318,11 +348,15 @@ class FilePrefetchBuffer {
   // curr_ represents the index for bufs_ indicating which buffer is being
   // consumed currently.
   uint32_t curr_;
+  // poll_index_ represents which buffer index to poll to get async results.
```

Comment:
In the code we are always consuming one buffer at a time. and it can be both buffer's have   async_read_in_progress set to true. In that case i don't know which buffer to poll in next call. Specially when data is overlapping.

Reply:
But I think you are right. Now I don't need poll_index_ because buffers are always sequential. Its curr_ and then second. Let me test it locally and see.